ios/RNSentry.mm

```diff
@@ -498,6 +503,46 @@ - (void)setEventEnvironmentTag:(SentryEvent *)event
     // the 'tracesSampleRate' or 'tracesSampler' option.
 }
 
+static NSString* const enabledProfilingMessage = @"Use SENTRY_PROFILING_ENABLED env to build with enabled profiling";
+
+RCT_EXPORT_SYNCHRONOUS_TYPED_METHOD(NSDictionary *, startProfiling)
+{
+#ifdef SENTRY_PROFILING_ENABLED
+    try {
+        facebook::hermes::HermesRuntime::enableSamplingProfiler();
+        return @{ @"started": @YES };
+    } catch (const std::exception& ex) {
```

Comment:
I'm not entirely sure how `std::exception` works on iOS but it's not used in Cocoa, the only reference is:
https://github.com/getsentry/sentry-cocoa/blob/b9b0f0a4b268f1ed4b30104486d690931dab63e1/Sources/SentryCrash/Recording/Monitors/SentryCrashMonitor_CPPException.cpp#L134
I know that exceptions don't work that well in iOS, maybe better to check with @brustolin or @philipphofmann here if this is necessary or if there's any other way to guard this function.

Reply:
Objc try/catch not that good, but sometimes is not possible to avoided it.
But this is C++ code, and I dont think this try/catch will be a problem.

Reply:
Gotcha, just in case, a bit of basic guidelines https://learn.microsoft.com/en-us/cpp/cpp/errors-and-exception-handling-modern-cpp?view=msvc-170#basic-guidelines

Reply:
This code can throw an exception if Hermes was compiled without a profiler.

https://github.com/krystofwoldrich/hermes/blob/4bcd62206c672c5b9674cab1399931ae8c275829/API/hermes/hermes.cpp#L1070-L1076